plugins/modules/win_pssession_configuration.ps1

```diff
@@ -203,6 +203,155 @@ function ConvertFrom-AnsibleOption {
+                $ht[$keyName] = ConvertTo-PsNative -Value $v
+            }
+            return $ht
+        }
+
+        # IEnumerable (but not string) -> array of parsed elements
+        if ($Value -is [System.Collections.IEnumerable] -and -not ($Value -is [string])) {
+            $out = @()
+            foreach ($e in $Value) { $out += (ConvertTo-PsNative -Value $e) }
+            return $out
+        }
+
+        return $Value
+    }
+}
+
+function ConvertTo-PsList {
+    <#
+    .SYNOPSIS
+    Normalize runtime objects into a PowerShell array of native elements.
+
+    .DESCRIPTION
+    Derived from the C# ParseList implementation in Ansible.Basic.cs.
+
+    .NOTES
+    - This is used for Ansible argument spec entries declared as "type='list'"
+        with "elements='raw'" so that module parameters arriving as CLR or
+        PSObject-wrapped values are made PowerShell-native before being passed
+        to PowerShell cmdlets.
+    #>
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true, ValueFromPipeline=$true)]
+        $Value
+    )
+
+    End {
+        if ($null -eq $Value) { return $null }
+        # Port of C# ParseList: perform .NET type checks to mirror behavior exactly.
+        $valueType = $Value.GetType()
+
+        # We'll collect parsed elements first, then coerce each element into the
+        # final shape (string or Hashtable) expected by New-PSSessionConfigurationFile.
+        $out = @()
+
+        # Unwrap PSObject if present
+        if ($Value -is [System.Management.Automation.PSObject] -and $Value.BaseObject -ne $null) {
+            $Value = $Value.BaseObject
+            $valueType = $Value.GetType()
+        }
+
+        if ($valueType.IsGenericType -and $valueType.GetGenericTypeDefinition().FullName -eq 'System.Collections.Generic.List`1') {
+            foreach ($e in $Value) { $out += (ConvertTo-PsNative -Value $e) }
+        }
+        elseif ($valueType.FullName -eq 'System.Collections.ArrayList') {
+            foreach ($e in $Value) { $out += (ConvertTo-PsNative -Value $e) }
+        }
+        elseif ($valueType.IsArray) {
+            foreach ($e in $Value) { $out += (ConvertTo-PsNative -Value $e) }
+        }
+        elseif ($valueType.FullName -eq 'System.String') {
+            $parts = $Value -split ',' | ForEach-Object { $_.Trim() }
+            foreach ($p in $parts) { $out += (ConvertTo-PsNative -Value $p) }
+        }
+        elseif ($valueType.FullName -eq 'System.Int32') {
+            $out += (ConvertTo-PsNative -Value $Value)
+        }
+        else {
+            throw "{0} cannot be converted to a list" -f $valueType.FullName
+        }
+
+        # Now coerce each parsed element into either a Hashtable (for dict-like
+        # values) or a string. Also flatten nested enumerables that may have been
+        # produced by ConvertTo-PsNative.
+        $final = @()
+        foreach ($e in $out) {
```

Comment:
Is there a reason that `ConvertTo-PsNative` can't handle all this logic, it seems to already? Seems wrong to do a second pass through on the values.